FogCreek/Glitch-Community
# Quixotic legend 1: componentize team users

## Links
* Remix link: https://quixotic-legend-1.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328406/;  also fixes https://glitch.manuscript.com/f/cases/3328633/

## GIF/Screenshots:
A team you can join
![Screen Shot 2019-05-29 at 5 20 40 PM](https://user-images.githubusercontent.com/938722/58592595-d0891380-8236-11e9-8b8c-8d0951c1f9c3.png)

A team you administrate 
![Screen Shot 2019-05-29 at 5 20 56 PM](https://user-images.githubusercontent.com/938722/58592596-d0891380-8236-11e9-98c4-3c91fe0f7dcd.png)

A team with lots of members
![Screen Shot 2019-05-29 at 5 21 19 PM](https://user-images.githubusercontent.com/938722/58592597-d0891380-8236-11e9-90c2-c7efe226572a.png)

Adding a user
![Screen Shot 2019-05-29 at 5 27 47 PM](https://user-images.githubusercontent.com/938722/58592748-2362cb00-8237-11e9-816c-4a64ad32e967.png)

## Changes:
* componentize AddTeamUserPop
* use Algolia for user search
* componentize TeamUsers
* make TeamUserInfoPop render a single user, not the whole list
* make PopoverSearch component
* Make components for ResultsList styles
* make toggleAndCall helper for popovers

## How To Test:
* log into https://quixotic-legend-1.glitch.me
* create a team
* clicking "Add" should show a user search list
* searching for a username should show a list of matching users
  * clicking on a user should invite them to your team
  * invited users should be visible in the teams list at 50% opacity
  * the other user should have received an invite
* searching for `@glitch.com` should allow you to whitelist email addresses from glitch.com
  * clicking the domain should add the glitch favicon to the list of team users
  * any user with that email address should see the 'join' button on your team page & can join team
* searching for a common free email address should (e.g. `@gmail.com`) should _not_ allow you to whitelist email addresses from gmail.com

## For future PRs
- use PopoverSearch, ResultsList components, toggleAndCall in other popovers
